File: LinguagemC/vetor_media.c
/*
Algoritmos Resolvidos em C
Prof.José Augusto Cintra - http://josecintra.com/blog

Enunciado: Leia as notas de 10 alunos e calcule a media geral.
           Depois verifique quais alunos possuem nota maior que a média
*/

#include <stdio.h>

int main() {
  
  float nota[10];
  float soma, media;
  const int n = 10;
  for (int i=0; i < n; i++) {
    printf("Digite a nota do aluno %d: ",i);
    scanf("%f",&nota[i]);
    soma += nota[i];
  }
  media = soma/10;
  printf("A media geral dos alunos e = %f\n",media);
  printf("Alunos que obtiveram notas maiores que a media:\n");
  for (int i=0; i < n; i++) {
    if (nota[i] > media) {
      printf("Aluno %d\n",i);
    }  
  }
}

File: LinguagemGO/vetorMedia.go
/*
Algoritmos Resolvidos em GOLANG

Prof.José Augusto Cintra 
http://josecintra.com/blog
https://github.com/JoseCintra/AlgoritmosResolvidos

Enunciado: Leia 10 notas e armazene em um vetor.
           Calcule a média geral e depois relacione todas as notas 
           maiores que a média
           

*/
package main

import "fmt"

func main() {
	const numNotas int = 10
	var (
		nota        [numNotas]float64
		soma, media float64 = 0, 0
	)

	for i := 0; i < numNotas; i++ {
		fmt.Printf("Informe a nota %d: ", i)
		fmt.Scan(&nota[i])
		soma += nota[i]
	}
	media = soma / float64(numNotas)
	fmt.Printf("Média = %f\n", media)
	fmt.Println("Notas Maiores do que a média:")
	for indice, valor := range nota {
		if valor > media {
			fmt.Printf("Nota %d = %f\n", indice, valor)
		}
	}
}
